src/auth/index.js

```diff
@@ -123,7 +124,7 @@ export default class Auth {
       }
 
       // If we get here, the state is valid - set auth appropriately.
-      return this.set(redirectState.ua);
+      return this.getLoggedInProviderType().then(providerType => this.set(redirectState.ua, providerType));
```

Comment:
[major] How does the ordering work here? `getLoggedInProviderType` wouldn't be able to return anything useful when it's the first time we're here and as a result, `set` will fail due to the restriction it has in place about empty types.